### PR TITLE
[CLI/test] Add better testing for wait operation and other middleware.

### PR DESCRIPTION
**Pull request by charlesg3**
**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*
Adds further testing for cli options.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/tests/pcluster/cli/test_delete_cluster.py [hunk 1]
```diff
@@ -81,6 +81,7 @@ def test_execut_with_wait(self, mocker):
         assert_that(delete_cluster_mock.call_args).is_length(2)  # this is due to the decorator on delete_cluster
         args_expected = {"region": None, "cluster_name": "cluster"}
         delete_cluster_mock.assert_called_with(**args_expected)
+        print(cf_waiter_mock.call_args.kwargs)
```
Comment (hanwen-pcluste) [inline cli/tests/pcluster/cli/test_delete_cluster.py]: You'd keep the print?

File: cli/tests/pcluster/cli/test_create_cluster.py [hunk 2]
```diff
@@ -148,7 +148,7 @@ def test_execute_with_wait(self, mocker, test_datadir):
             "create_cluster_request_content": {"clusterName": "cluster", "clusterConfiguration": ""},
         }
         create_cluster_mock.assert_called_with(**expected_args)
-        assert_that(cf_waiter_mock.call_args.kwargs).is_equal_to({"StackName": "cluster"})
+        assert_that(cf_waiter_mock.call_args[1]).is_equal_to({"StackName": "cluster"})
```
Comment (hanwen-pcluste) [inline cli/tests/pcluster/cli/test_create_cluster.py:151]: Is the backward compatibility of Python?

File: cli/tests/pcluster/cli/test_update_cluster.py [hunk 3]
```diff
@@ -68,6 +70,87 @@ def test_invalid_args(self, args, error_message, run_cli, capsys, test_datadir):
...
+        cf_waiter_mock = mocker.patch("botocore.waiter.Waiter.wait")
+        mock_aws_api(mocker)
+
+        path = str(test_datadir / "config.yaml")
+        command = ["update-cluster", "--cluster-name", "cluster", "--cluster-configuration", path, "--wait"]
+        out = run(command)
+
+        expected = wire_translate(response)
+        assert_that(out).is_equal_to(expected)
+        assert_that(update_cluster_mock.call_args).is_length(2)  # this is due to the decorator on list_clusters
```
Comment (hanwen-pcluste) [inline cli/tests/pcluster/cli/test_update_cluster.py:140]: What do you mean by the comment?

File: cli/tests/pcluster/cli/test_delete_cluster.py (hunk 1 above)
Comment (charlesg3) [reply cli/tests/pcluster/cli/test_delete_cluster.py]: No that's a mistake.

Comment (charlesg3) [reply cli/tests/pcluster/cli/test_delete_cluster.py]: I was having trouble with previous versions of python that I had to debug with the CI pipeline...

Comment (charlesg3) [reply cli/tests/pcluster/cli/test_delete_cluster.py]: (But it should be removed now?)

File: cli/tests/pcluster/cli/test_update_cluster.py (hunk 3 above)
Comment (charlesg3) [reply cli/tests/pcluster/cli/test_update_cluster.py:140]: Removed.

**Review by hanwen-pcluste — APPROVED** [review]

**Pull request merged**